### ISSUE TITLE
Support customized issuer values for GitHub Enterprise Cloud

### ISSUE THREAD

**Issue by isometry** [actions/attest-build-provenance#222]
The action is currently incompatible with any organisation belonging to a GitHub Enterprise Cloud instance that has followed the hardening guidance to [customize the issuer value][1]:

```
Error: Failed to get ID token: unexpected "iss" claim value
```

With the [GitHub Actions OIDC Custom Issuer Policy][2] enabled (i.e. `include_enterprise_slug=true`), the Enterprise Cloud issuer becomes `https://token.actions.githubusercontent.com/${enterpriseSlug}`.

Ideally, the action would dynamically retrieve the correct `enterpriseSlug` given the runtime context; alternatively it should support an optional `enterpriseSlug` input parameter to be suffixed to the calculated [issuer][3].

[1]: https://docs.github.com/en/enterprise-cloud@latest/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#customizing-the-issuer-value-for-an-enterprise
[2]: https://docs.github.com/en/enterprise-cloud@latest/rest/actions/oidc?apiVersion=2022-11-28#set-the-github-actions-oidc-custom-issuer-policy-for-an-enterprise
[3]: https://github.com/actions/attest-build-provenance/blob/main/src/main.ts#L30

**Comment by bdehamer** [comment]
Definitely something we need to fix. 

I don't think there is a good way to determine whether the issuer customization is enabled for a given account, so will probably end-up with an `include_enterprise_slug` input param.

**Comment by bdehamer** [comment]
This should be resolved with the [v1.4.3](https://github.com/actions/attest-build-provenance/releases/tag/v1.4.3) release. We didn't end-up adding a new input parameter -- it should just work whether or not you have customized the issuer URL for your organization.
